File: app/src/main/java/com/example/sony/newsapi/screen/article/ArticleContract.java
package com.example.sony.newsapi.screen.article;

public interface ArticleContract {

    interface View {
        void openPage(String url);

        void showTitleOfActionBar(String title);
        void showProgressBar();
        void hideProgressBar();
        void showToast(String message);
    }

    interface Presenter {
        void loadUrl(int index);

        void onPageStarted();
        void onPageFinished();
        void onReceivedError();
    }
}

File: app/src/main/java/com/example/sony/newsapi/screen/article/ArticleContract.kt
package com.example.sony.newsapi.screen.article

interface ArticleContract {

    interface View {
        fun openPage(url: String)

        fun showTitleOfActionBar(title: String)
        fun showProgressBar()
        fun hideProgressBar()
        fun showToast(message: String)
    }

    interface Presenter {
        fun loadUrl(index: Int)

        fun onPageStarted()
        fun onPageFinished()
        fun onReceivedError()
    }
}
